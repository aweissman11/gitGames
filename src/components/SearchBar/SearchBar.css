.search-bar {
  display: flex;
  justify-content: space-between;
  position: absolute;
  top: 0;
  left: 0;
  width: 100vw;
  z-index: 9999999;
}

.git-games {
  display: flex;
  justify-content: left;
  line-height: 70px;
}

.title {
  color: blueviolet;
  font-size: 45px;
}

.big-g {
  font-size: 1em;
  font-weight: bold;
}

.logo {
  fill: blueviolet;
  width: 50px;
  height: 50px;
  margin: 10px;
}

