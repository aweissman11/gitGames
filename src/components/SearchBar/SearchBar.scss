@import '../../App.scss';

.search-bar {
  display: flex;
  height: 80px;
  justify-content: space-between;
  position: absolute;
  top: 0;
  left: 0;
  width: 100vw;
  z-index: 99;
}

.logo-box {
  display: flex;
  justify-content: space-around;
  align-items: center;
}

.git-games {
  display: flex;
  justify-content: left;
  line-height: 70px;
}

.title {
  color: $my-white;
  font-size: 45px;
}

.big-g {
  font-size: 1em;
  font-weight: bold;
}

.logo {
  fill: $my-white;
  width: 50px;
  margin-left: 10px;
  height: 50px;
  position: relative;
  top: 14px;
  left: 5px;
}

.logo:hover {
  // fill: rgb(220, 220, 220);
  filter: brightness(.8)
}

.logo:active {
  filter: brightness(.5)
}

@media (max-width: 640px) {
  .user-data {
    display: flex;
    justify-content: flex-end;
    min-width: 50px;
    // justify-items: flex-end;
  }

  .header-avatar-box {
  }
  
  .username {
    display: none;
  }
}

// @media (max-width: 850px) {
//   .force-container {
//     height: 100%;
//     width: 132vw;
//   }
//   .search-bar {
//     height: 70px;
//   }

//   .git-games {
//     flex-direction: column;
//     height: 70px;
//   }

//   .git-games a, .git-games p {
//     height: 30px;
//   }

//   .title {
//     margin-top: 10px;
//     font-size: 1.5em;
//     line-height: 30px;
//   }

//   .big-g {
//     font-size: 1.2em;
//   }

//   .logo-box {
//     position: relative;
//   }

//   .logo {
//     height: 25px;
//     width: 25px;
//     position: absolute;
//     top: 2px;
//     left: 5px;
//   }

//   .user-data {
//     padding-top: 0;
//     margin-top: 0;
//     /* height: 30px; */
//   }

//   .username {
//     font-size: 1em;
//   }

//   .user-data .loading-span {
//     height: 40px;
//     line-height: 40px;
//   }
// }