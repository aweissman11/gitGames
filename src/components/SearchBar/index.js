import React from 'react';
import { Link } from 'react-router-dom'

import SearchUsers from '../SearchUsers';
import UserData from '../../containers/UserData';

import { ReactComponent as LogoSvg } from '../../assets/Octicons-mark-github.svg';

import './SearchBar.css';

export const SearchBar = (props) => (
  <div className='search-bar'>
      <div className='git-games'>
        <div className='logo-box'>
          <Link to='/'>
            <LogoSvg className='logo' />
          </Link>
          <p className='title'><span className='big-g'>G</span>it<span className='big-g'>G</span>ames</p>
        </div>
        <UserData loginPage={props.loginPage}/>
      </div>
    <SearchUsers />
  </div>
)

export default SearchBar;