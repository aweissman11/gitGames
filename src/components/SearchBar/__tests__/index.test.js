import React from 'react';
import { shallow } from 'enzyme';

import SearchBar from '../index';

describe('SearchBar', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<SearchBar loginPage={{user: 'me'}} />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  
})
