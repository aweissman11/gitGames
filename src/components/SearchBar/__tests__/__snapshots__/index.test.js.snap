// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar
    loginPage={
      Object {
        "user": "me",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="git-games"
        >
          <div
            className="logo-box"
          >
            <p>
              <Link
                replace={false}
                to="/"
              >
                <ReactComponent
                  className="logo"
                />
              </Link>
            </p>
            <p>
              <Connect(SearchInput) />
            </p>
          </div>
        </div>,
        <div
          className="search-bar-user-data"
        >
          <Connect(UserData)
            loginPage={
              Object {
                "user": "me",
              }
            }
          />
        </div>,
      ],
      "className": "search-bar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="logo-box"
          >
            <p>
              <Link
                replace={false}
                to="/"
              >
                <ReactComponent
                  className="logo"
                />
              </Link>
            </p>
            <p>
              <Connect(SearchInput) />
            </p>
          </div>,
          "className": "git-games",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                <Link
                  replace={false}
                  to="/"
                >
                  <ReactComponent
                    className="logo"
                  />
                </Link>
              </p>,
              <p>
                <Connect(SearchInput) />
              </p>,
            ],
            "className": "logo-box",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/"
                >
                  <ReactComponent
                    className="logo"
                  />
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <ReactComponent
                    className="logo"
                  />,
                  "replace": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "className": "logo",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Connect(SearchInput) />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "p",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(UserData)
            loginPage={
              Object {
                "user": "me",
              }
            }
          />,
          "className": "search-bar-user-data",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "loginPage": Object {
              "user": "me",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="git-games"
          >
            <div
              className="logo-box"
            >
              <p>
                <Link
                  replace={false}
                  to="/"
                >
                  <ReactComponent
                    className="logo"
                  />
                </Link>
              </p>
              <p>
                <Connect(SearchInput) />
              </p>
            </div>
          </div>,
          <div
            className="search-bar-user-data"
          >
            <Connect(UserData)
              loginPage={
                Object {
                  "user": "me",
                }
              }
            />
          </div>,
        ],
        "className": "search-bar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="logo-box"
            >
              <p>
                <Link
                  replace={false}
                  to="/"
                >
                  <ReactComponent
                    className="logo"
                  />
                </Link>
              </p>
              <p>
                <Connect(SearchInput) />
              </p>
            </div>,
            "className": "git-games",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  <Link
                    replace={false}
                    to="/"
                  >
                    <ReactComponent
                      className="logo"
                    />
                  </Link>
                </p>,
                <p>
                  <Connect(SearchInput) />
                </p>,
              ],
              "className": "logo-box",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    replace={false}
                    to="/"
                  >
                    <ReactComponent
                      className="logo"
                    />
                  </Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <ReactComponent
                      className="logo"
                    />,
                    "replace": false,
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "className": "logo",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Connect(SearchInput) />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "p",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(UserData)
              loginPage={
                Object {
                  "user": "me",
                }
              }
            />,
            "className": "search-bar-user-data",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "loginPage": Object {
                "user": "me",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
